### PR TITLE
Only showing "can't save changes" alert if changes were actually made.

### LINKED ISSUE #11312: Remove "Could not save changes" alert if no changes were made.
Steps:

- Be offline
- Go to sharing screen
- Go to manage
- Go back 

See the alert. 
IMG_3599

### PR DESCRIPTION
**Pull request by yaelirub**
Fixes #11312 

Before:
sharing_offline_before

After:
sharing_offline_2

To test:
1. Be offline
2. Go to sharing
3. Go to manage
4. Don't change anything
5. Go back
6. See that the alert is not displayed

1. Be online
2. Go to sharing
3. Go to manage 
4. Make changes 
5. Make sure they are saved

Update release notes:

- [ ] If there are user facing changes, I have added an item to `RELEASE-NOTES.txt`.

### CODE REVIEW

File: WordPress/Classes/ViewRelated/Blog/SharingButtonsViewController.swift [hunk 1]
```diff
@@ -520,7 +533,7 @@ import WordPressShared
     ///
     /// - Parameter refresh: True if the tableview should be reloaded.
     ///
-    @objc func saveBlogSettingsChanges(_ refresh: Bool) {
+    private func saveBlogSettingsChanges(_ refresh: Bool) {
```
Comment (shiki) [inline WordPress/Classes/ViewRelated/Blog/SharingButtonsViewController.swift]: Nice cleanup 👍

File: Podfile.lock [hunk 2]
```diff
@@ -231,7 +231,7 @@ PODS:
   - ZendeskSDK/UI (2.2.0):
     - ZendeskSDK/Core
     - ZendeskSDK/Providers
-  - ZIPFoundation (0.9.8)
+  - ZIPFoundation (0.9.9)
```
[the remark this answers is not in the document]
Comment (yaelirub) [reply Podfile.lock]: I used bundle exec pod update. Had some pod issues yesterday switching back and forth to old release branches. I usually do bundle exec pod install. 
Should I remove the update from this PR?

File: WordPress/Classes/ViewRelated/Blog/SharingButtonsViewController.swift (hunk 1 above)
Comment (yaelirub) [reply WordPress/Classes/ViewRelated/Blog/SharingButtonsViewController.swift]: I'm updating this PR with more cleanup.

**Comment by yaelirub** [comment]
> I was wondering if these should be Notices instead of an alert. It's fairly annoying

I agree but not sure that notice is the way to go here. How about we remove the alert from the page itself and only leave the one when existing the page and only if changes were made?

>I have found that the ordering is not considered as a "did make changes" and the app does not try to save it. Is that something that should be fixed with this PR?

Absolutely should be part of this PR. Updating.

File: WordPress/Classes/ViewRelated/Blog/SharingButtonsViewController.swift [hunk 3]
```diff
@@ -770,21 +750,65 @@ import WordPressShared
         navigationController?.pushViewController(controller, animated: true)
     }
 
+    /// Represents a section in the sharinging management table view.
```
Comment (yaelirub) [inline WordPress/Classes/ViewRelated/Blog/SharingButtonsViewController.swift]: These classes were just moved

**Comment by yaelirub** [comment]
@shiki , @diegoreymendez updated and ready for another look.

File: Podfile.lock (hunk 2 above)
[the remark this answers is not in the document]
Comment (shiki) [reply Podfile.lock]: It's a minor version change so I hope it's fine. I asked Platform if we have a process for this.

**Comment by diegoreymendez** [comment]
I'm currently not seeing the alert at all, wondering if something else is at play here.  The test shown below is executed offline (in fact when I go back into the "Manage" screen, my changes to the settings haven't been saved).

noAlert

**Comment by yaelirub** [comment]
@diegoreymendez , on simulator or device?

**Comment by diegoreymendez** [comment]
@yaelirub - Simulator, is that what's causing the difference?

**Comment by yaelirub** [comment]
@diegoreymendez , I was asking because I am unable to test on device atm.
The alert is being displayed for me. I think that the simulator doesn't behave consistently when turning the wifi on mac off to test offline. Can . you possible test on device and see if you have the same issue?

@shiki , can you see the alert?

**Comment by shiki** [comment]
@yaelirub I retested this after @diegoreymendez mentioned he couldn't see the alert. I couldn't make the alert show at that time. 😞I'll try again.

**Comment by yaelirub** [comment]
Ok. Thanks @shiki . Jus t a reminder that the alert will only show if you actually try to make changes in sharing sources. Meaning it wont show it you only toggle the edit and more buttons but not the list items under them.

**Comment by shiki** [comment]
Tested this again and can confirm that `saveButtonChanges` is called when leaving the page and the alert is shown. Tested on iPad Pro. iOS 12.1.4.

ScreenRecording_04-15-2019 15-06-58 2019-04-15 15_17_13

LGTM!

**Pull request merged**